app/assets/javascripts/application.js

```diff
@@ -351,6 +351,12 @@ $(function() {
   });
 });
 
+function check_for_unavailable_puppetclasses() {
+  if ($('#puppet_klasses #selected_classes .unavailable').size() > 0) {
+    notify('<span>' + __('Some Puppet Classes are unavailable in the selected environment') + '</span>', 'warning')
```

Comment:
BUT. I do think that "Puppet class" would be the correct way to write it, as "Puppet" is a "brand" and "class" is just another word.